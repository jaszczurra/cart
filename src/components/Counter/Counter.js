import React from 'react'
import bemCx from 'bem-modifiers'
import PropTypes from 'prop-types'

const Counter = ({ min, max, value, isBlocked, handleChange }) => {

  const handleIncrease = () => {
    value < max && !isBlocked && handleChange(value + 1)
  }

  const handleDecrease = () => {
    value > min && !isBlocked && handleChange(value - 1)
  }

  return (
    <div className={bemCx('counter', { 'disabled': isBlocked } )}>
      <div className='counter__button' onClick={handleDecrease}>-</div>
      <div className='counter__text'>{`Obecnie masz ${value} sztuk produktu`}</div>
      <div className='counter__button' onClick={handleIncrease}>+</div>
    </div>
  )
}

Counter.propTypes = {
  min: PropTypes.number,
  max: PropTypes.number,
  value: PropTypes.number.isRequired,
  isBlocked: PropTypes.bool,
  handleChange: PropTypes.func.isRequired,
}

Counter.defaultProps = {
  min: 0,
  max: 999,
}

export { Counter }
