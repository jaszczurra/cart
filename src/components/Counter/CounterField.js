import React from 'react'
import get from 'lodash/get'
import PropTypes from 'prop-types'

import { Counter } from './Counter'

const CounterField = ({ form, field, min, max, isBlocked, onChangeCallback }) => {

  const handleChange = value => {
    form.setFieldValue(field.name, value)
    onChangeCallback && onChangeCallback()
  }

  return (
    <Counter
      min={min}
      max={max}
      isBlocked={isBlocked}
      value={get(form.values, field.name)}
      handleChange={handleChange}
    />
  )
}

CounterField.propTypes = {
  form: PropTypes.object,
  field: PropTypes.object,
  min: PropTypes.number,
  max: PropTypes.number,
  isBlocked: PropTypes.bool,
  onChangeCallback: PropTypes.func,
}

export { CounterField }
