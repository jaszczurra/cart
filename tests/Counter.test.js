import React from 'react'
import renderer from 'react-test-renderer'

import { Counter } from '../src/components/Counter/Counter'

describe('Counter', () => {

  it('Snapshot', () => {
    const component = renderer.create(<Counter value={5} handleChange={() => {}} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
