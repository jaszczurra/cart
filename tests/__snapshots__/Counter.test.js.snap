// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot 1`] = `
<div
  className="counter"
>
  <div
    className="counter__button"
    onClick={[Function]}
  >
    -
  </div>
  <div
    className="counter__text"
  >
    Obecnie masz 5 sztuk produktu
  </div>
  <div
    className="counter__button"
    onClick={[Function]}
  >
    +
  </div>
</div>
`;
